### PR TITLE
Allow send_email method to accept email_reply_to_id 

### PR DESCRIPTION
**Pull request by richardc0**
Updated the client to allow the client to pass in an email_reply_to_id
when sending a notification.

- Added Integration Tests to Client to test the new functionality
- Updated Client Code - Interface and the implementation
- Updated code to allow HTTP connections as well as HTTPS connections for easier local testing

### CODE REVIEW

**Comment by tamasperger** [comment]
Additionally to pom.xml, please update the version in the following files, too:
src/main/resources/application.properties
src/test/java/uk/gov/service/notify/NotificationClientTest.java
CHANGELOG.md

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Then add the Maven dependency to your project.
     <dependency>
         <groupId>uk.gov.service.notify</groupId>
         <artifactId>notifications-java-client</artifactId>
-        <version>3.4.0-RELEASE</version>
+        <version>3.5.0-RELEASE</version>
```
Comment (tamasperger) [inline README.md]: I would consider changing the version to 3.6.0-RELEASE here, too.
The same goes for line 63 (in the Gradle section below).

**Comment by venusbb** [comment]
Can you put the comment on using http rather than https in the PR description pls?

File: src/test/java/uk/gov/service/notify/ClientIntegrationTestIT.java [hunk 2]
```diff
@@ -279,5 +279,59 @@ private void assertNotificationWhenSms(Notification notification) {
...
+                System.getenv("EMAIL_TEMPLATE_ID"),
+                System.getenv("FUNCTIONAL_TEST_EMAIL"),
+                personalisation,
+                uniqueName,
+                System.getenv("EMAIL_REPLY_TO_ID"));
+
+        assertNotificationEmailResponse(response, uniqueName);
+
+        Notification notification = client.getNotificationById(emailResponse.getNotificationId().toString());
+        assertNotification(notification);
```
Comment (venusbb) [inline src/test/java/uk/gov/service/notify/ClientIntegrationTestIT.java]: - Do you need to assert that the notification is sent with the reply_to_address other than the default?
- You may also want to add the new field in assertNotification()

**Pull request merged**